editor/gui/scene_tree_editor.cpp

```diff
@@ -217,7 +218,52 @@ void SceneTreeEditor::_toggle_visible(Node *p_node) {
 	}
 }
 
-void SceneTreeEditor::_add_nodes(Node *p_node, TreeItem *p_parent) {
+static uint32_t _hash_node(Node *p_node) {
+	uint32_t hash = p_node->get_name().hash();
+
+	hash = hash_djb2_one_32(p_node->is_unique_name_in_owner(), hash);
+
+	// If the warnings changed we need to change the tooltip and button.
+	PackedStringArray warnings = p_node->get_configuration_warnings();
+	for (const String &s : warnings) {
+		hash = hash_djb2_one_32(s.hash(), hash);
+	}
+
+	// If metadata changed we might need to update lock/group buttons.
+	List<StringName> metadata;
+	p_node->get_meta_list(&metadata);
+	for (const StringName &s : metadata) {
+		hash = hash_djb2_one_32(s.hash(), hash);
+	}
+
+	// If the script changed we need to update the tooltip and button.
+	hash = hash_djb2_one_32(p_node->get_script().hash(), hash);
+
+	// For signals and groups we only care if the number changed.
```

Comment:
Group names are listed in the tooltip.
![image](https://github.com/user-attachments/assets/9250fc8d-ea22-4899-bd88-878cba5ff388)
If you rename a group (rare, but can happen), it becomes outdated.

Reply:
Okay, I can just hash the group names then instead. It shouldn't matter that much for performance unless loads of nodes are in loads of groups.

Reply:
This matches current behavior on master, as discussed we'll do a separate bugfix PR.